qiskit/compiler/assembler.py

```diff
@@ -189,6 +189,16 @@ def assemble(
     )
 
 
+# Note for future: this method is used in `BasicSimulator` and may need to be kept past the
+# `assemble` removal deadline (2.0). If it is kept (potentially in a different location),
+# we will need an alternative for the backend.configuration() access that currently takes
+# place in L566 (`parse_circuit_args`) and L351 (`parse_common_args`)
+# because backend.configuration() is also set for removal in 2.0.
+# The ultimate goal will be to move away from relying on any kind of `assemble` implementation
+# because of how tightly coupled it is to these legacy data structures. But as a transition step,
+# given that we would only have to support the subcase of `BasicSimulator`, we could probably just
+# inline the relevant config values that are already hardcoded in the basic simulator configuration
+# generator.
```

Comment:
FWIW, it should not be hard to rewrite the `run_experiment` method of basic simulator to avoid needing qobj. The assemble usage in there is kind of a weird legacy from `BaseBackend` which required a qobj get passed to the `run()` method.